src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3132,7 +3132,7 @@ final void endMessage() throws SQLServerException {
     // the client MUST send the next packet with both ignore bit (0x02) and EOM bit (0x01)
     // set in the status to cancel the request.
     final boolean ignoreMessage() throws SQLServerException {
-        if (packetNum > 0) {
+        if (packetNum > 0 || TDS.PKT_BULK == this.tdsMessageType) {
```

Comment:
Do we need this TDS.PKT_BULK == this.tdsMessageType check? In SQLServerBulkCopy::sendBulkCopyCommand, I see that we start the request by sending the TDS.PKT_BULK packet. However, before we get there, we already write PKT_QUERY packet, which increments the packetNum to >0. Doesn't this mean we don't need to change this if statement?

Reply:
`packetNum` is reset every time you start a new request, `TDS.PKT_BULK` in this case, but we still need to cancel `TDS.PKT_QUERY`.